### PR TITLE
Device info page

### PR DESCRIPTION
**Pull request by borntocompile**
To create the device info page I had to rework the way the cloud info page works. It now programmatically creates the title and its value  on Awake by inheriting the `AbstractInfoSubPage`.

```c#
AddElement("Cloud Build Target Name", BuildManifest.CloudBuildTargetName);
```

The Device info page consists of a list of useful device info.
* Device Name
* Device Model
* Device Type
* Operating System
* Operating System Family
* System Memory Size
* Processor Type
* Processor Count
* Screen Resolution
* Screen Aspect Ratio
* Graphics Device Name
* Graphics Device Type
* Graphics Device Vendor
* Graphics Memory Size
* Graphics Shader Level
* Graphics Multi Threaded
* Battery Status
* Battery Level
* Supports Vibration
* Supports Shadows
* Supports Instancing
* Supports Location Services
* Supports Accelerometer
* Supports Gyroscope
* Supports Audio

And looks like this:
screen shot 2018-06-04 at 10 08 41

### CODE REVIEW

File: Scripts/InfoPage/AbstractInfoSubPage.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using UnityEngine;
+
+namespace DUCK.DebugMenu.InfoPage
+{
+	public class AbstractInfoSubPage : MonoBehaviour
+	{
+		private const int DEFAULT_CHILDREN = 2;
```
Comment (kkjamie) [inline Scripts/InfoPage/AbstractInfoSubPage.cs:7]: I'd probably leave an empty line between const and member fields

File: Scripts/InfoPage/DeviceInfoManifest.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	{
+		public string DeviceName { get; private set; }
+		public string DeviceModel { get; private set; }
+		public string DeviceType { get; private set; }
+		public string OperatingSystem { get; private set; }
+		public string OperatingSystemFamily { get; private set; }
+		public string SystemMemorySize { get; private set; }
+		public string ProcessorType { get; private set; }
+		public string ProcessorCount { get; private set; }
+		public string ScreenResolution { get { return string.Format("{0}x{1} ({2})", Screen.width, Screen.height, Screen.height > Screen.width ? "Portrait" : "Landscape"); } }
```
Comment (kkjamie) [inline Scripts/InfoPage/DeviceInfoManifest.cs:16]: Can we pull DPI as well from screen ?

**Review by kkjamie — APPROVED** [review]

File: Scripts/InfoPage/AbstractInfoSubPage.cs (hunk 1 above)
Comment (borntocompile) [reply Scripts/InfoPage/AbstractInfoSubPage.cs:7]: Done

File: Scripts/InfoPage/DeviceInfoManifest.cs (hunk 2 above)
Comment (borntocompile) [reply Scripts/InfoPage/DeviceInfoManifest.cs:16]: Done

**Pull request merged**